toolchain/check/context.h

```diff
@@ -208,22 +212,32 @@ class Context {
   auto AddNameToLookup(SemIR::NameId name_id, SemIR::InstId target_id) -> void;
 
   // Performs name lookup in a specified scope for a name appearing in a
-  // declaration, returning the referenced instruction. If scope_id is invalid,
-  // uses the current contextual scope.
+  // declaration. If found, returns the referenced instruction and false. If
+  // poisoned, returns invalid instructions and true. poisoned. If scope_id is
+  // invalid, uses the current contextual scope.
+  // TODO: For poisoned names, return the poisoning instruction.
   auto LookupNameInDecl(SemIR::LocId loc_id, SemIR::NameId name_id,
-                        SemIR::NameScopeId scope_id) -> SemIR::InstId;
+                        SemIR::NameScopeId scope_id)
+      -> std::pair<SemIR::InstId, bool>;
 
   // Performs an unqualified name lookup, returning the referenced instruction.
   auto LookupUnqualifiedName(Parse::NodeId node_id, SemIR::NameId name_id,
                              bool required = true) -> LookupResult;
 
+  struct LookupNameInExactScopeResult {
+    SemIR::InstId inst_id;
+    SemIR::AccessKind access_kind;
+    bool is_poisoned = false;
```

Comment:
Is this default necessary, or could it be removed?

Reply:
I think it depends on how the struct is initialized, so this is safer.
If in some function definition you just do
`LookupNameInExactScopeResult result;`
Then I think nothing guarantees `is_poisoned` would be false.